examples/bond/test/bond-token-flow.js

```diff
@@ -0,0 +1,424 @@
+    const appArgs = ['str:update_issuer_address', convert.addressToPk(issuerLsigAddress)]; // converts algorand address to Uint8Array
+
+    const appCallParams = {
+      type: types.TransactionType.CallNoOpSSC,
+      sign: types.SignType.SecretKey,
+      fromAccount: appManager.account,
+      appID: applicationId,
+      payFlags: {},
+      appArgs: appArgs
+    };
+    runtime.executeTx(appCallParams);
+
+    // verify issuer address
+    assert.isDefined(applicationId);
+    assert.deepEqual(getGlobal('issuer_address'), convert.addressToPk(issuerLsigAddress));
+
+    // opt-in to app
+    runtime.optInToApp(appManager.address, applicationId, {}, {});
+    runtime.optInToApp(issuerAddress.address, applicationId, {}, {});
+
+    syncAccounts();
+    assert.isDefined(appManager.appsLocalState.get(applicationId));
+    assert.isDefined(issuerAddress.appsLocalState.get(applicationId));
+
+    runtime = optIn(runtime, lsig, currentBondIndex, appManager);
+
+    // Issue tokens to issuer from bond token creator
+    let groupTx = [
+      // Bond asa transfer to issuer's address
+      {
+        type: types.TransactionType.TransferAsset,
+        sign: types.SignType.SecretKey,
+        fromAccount: bondTokenCreator.account,
+        toAccountAddr: issuerLsigAddress,
+        amount: 1e6,
+        assetID: currentBondIndex,
+        payFlags: { }
+      },
+      // call to bond-dapp
+      {
+        type: types.TransactionType.CallNoOpSSC,
+        sign: types.SignType.SecretKey,
+        fromAccount: bondTokenCreator.account,
+        appID: applicationId,
+        payFlags: {},
+        appArgs: ['str:issue']
+      }
+    ];
+    runtime.executeTx(groupTx);
+
+    syncAccounts();
+    assert.equal(issuerAddress.getAssetHolding(currentBondIndex)?.amount, 1000000n);
+
+    // epoch_0 elon buys 10 bonds
+    runtime.optIntoASA(currentBondIndex, elon.address, {});
+    runtime.optInToApp(elon.address, applicationId, {}, {});
+    let amount = 10;
+    let algoAmount = amount * issue;
+
+    groupTx = [
+      // Algo transfer from elon to issuer
+      {
+        type: types.TransactionType.TransferAlgo,
+        sign: types.SignType.SecretKey,
+        fromAccount: elon.account,
+        toAccountAddr: issuerLsigAddress,
+        amountMicroAlgos: algoAmount,
+        payFlags: { totalFee: 2000 }
+      },
+      // Bond token transfer from issuer's address
+      {
+        type: types.TransactionType.TransferAsset,
+        sign: types.SignType.LogicSignature,
+        fromAccountAddr: issuerLsigAddress,
+        lsig: lsig,
+        toAccountAddr: elon.address,
+        amount: 10,
+        assetID: currentBondIndex,
+        payFlags: { totalFee: 0 }
+      },
+      // call to bond-dapp
+      {
+        type: types.TransactionType.CallNoOpSSC,
+        sign: types.SignType.SecretKey,
+        fromAccount: elon.account,
+        appID: applicationId,
+        payFlags: { totalFee: 1000 },
+        appArgs: ['str:buy']
+      }
+    ];
```

Comment:
ok, adding a function for issue and buy as well.

Reply:
and transaction parameters functions in scripts

Reply:
done, added groupTx functions for issue and buy, haven't added for createDex because it needs 10 parameters to make that transaction, I think it won't be much clear when we use function with 10 parameters while testing

Reply:
added function with 10 parameters